packages/ocapn/test/client.test.js

```diff
@@ -0,0 +1,136 @@
+// @ts-check
+
+/** @typedef {import('../src/codecs/components.js').OcapnLocation} OcapnLocation */
+/** @typedef {import('../src/client/types.js').Client} Client */
+
+import test from '@endo/ses-ava/prepare-endo.js';
+import { E } from '@endo/eventual-send';
+import { Far } from '@endo/marshal';
+import { isPromise } from '@endo/promise-kit';
+import { makeTcpNetLayer } from '../src/netlayers/tcp-test-only.js';
+import { makeClient } from '../src/client/index.js';
+import { testWithErrorUnwrapping } from './_util.js';
+import { encodeSwissnum } from '../src/client/util.js';
+
+/**
+ * @param {string} debugLabel
+ * @param {() => Map<string, any>} [makeDefaultSwissnumTable]
+ * @returns {Promise<{ client: Client, location: OcapnLocation }>}
+ */
+const makeTestClient = async (debugLabel, makeDefaultSwissnumTable) => {
+  const client = makeClient({
+    debugLabel,
+    swissnumTable: makeDefaultSwissnumTable && makeDefaultSwissnumTable(),
+  });
+  const tcpNetlayer = await makeTcpNetLayer({ client });
+  client.registerNetlayer(tcpNetlayer);
```

Comment:
I’m sure you’ve got an eye on how this integrates with the pet dæmon. We can add *and remove* netlayers at runtime? We probably ought to synchronously disconnect all living sessions for a netlayer when it’s unregistered.